lib/models/event.js

```diff
@@ -189,5 +189,28 @@ module.exports.MatrixEvent.prototype = {
 
     getUnsigned: function() {
         return this.event.unsigned || {};
-    }
+    },
+
+    /**
+     * Update the content of an event in the same way it would be by the server
+     * if it were redacted before it was sent to us
+     *
+     * @param {Object} the raw event causing the redaction
+     */
+    makeRedacted: function(redaction_event) {
+        if (!this.event.unsigned) {
+            this.event.unsigned = {};
+        }
+        this.event.unsigned.redacted_because = redaction_event;
+        this.event.content = {};
```

Comment:
Is this sufficient?  http://matrix.org/docs/spec/r0.0.1/client_server.html#redactions says: 

Upon receipt of a redaction event, the server should strip off any keys **not** in the following list:
 * event_id
 * type
 * room_id
 * user_id
 * state_key
 * prev_state
 * content

The content object should also be stripped of all keys, unless it is one of one of the following event types:

 * m.room.member allows key membership
 * m.room.create allows key creator
 * m.room.join_rules allows key join_rule
 * m.room.power_levels allows keys ban, events, events_default, kick, redact, state_default, users, users_default.
 * m.room.aliases allows key aliases

I have used redactions quite a few times on things like membership change events in order to redact offensive displaynames - is this going to be sufficient?